component/src/main/java/org/ballerinalang/sap/utils/SapUtils.java

```diff
@@ -260,53 +253,51 @@ public static Properties getDestinationProperties(Struct destinationConfig) {
      * @param sapService that contains resource map
      * @param error      {@link BError} instance which contains the error details
      */
-    public static void invokeOnError(Map<String, Resource> sapService, CallableUnitCallback callback, String error) {
+    public static void invokeOnError(Map<String, Resource> sapService, CallableUnitCallback callback, String error,
+                                     Context context) {
         try {
             Resource errorResource = sapService.get(RESOURCE_ON_ERROR);
-            log.info("invokeOnError : " + sapService.get(RESOURCE_ON_ERROR).getName());
-            BValue[] params = getOnErrorResourceSignature(errorResource, error);
-            Executor.submit(errorResource, callback, null, null, params);
+            Executor.submit(errorResource, callback, null, null, createError(context, error));
         } catch (Throwable e) {
             log.error("Error while executing onError resource", e);
         }
     }
 
-    private static BValue[] getOnErrorResourceSignature(Resource errorResource, String msg) {
-        ProgramFile programFile = errorResource.getResourceInfo().getPackageInfo().getProgramFile();
-        BError error = createSapError(programFile, msg);
-        return new BValue[] { error };
-    }
-
     /**
-     * Creates an error message.
+     * Invoke the 'onMessage' resource.
      *
-     * @param programFile ProgramFile which is used
-     * @param errMsg      the cause for the error
-     * @return an error which will be propagated to ballerina user
+     * @param output      Required for the resource
+     * @param sapResource The resource
+     * @param callback    Callback to return the response values
+     * @param context     Current context instance
      */
-    private static BError createSapError(ProgramFile programFile, String errMsg) {
-        BMap<String, BValue> errorRecord = BLangConnectorSPIUtil
-                .createBStruct(programFile, SAP_NATIVE_PACKAGE, CONSUMER_STRUCT_NAME);
-        errorRecord.put("message", new BString(errMsg));
-        return BLangVMErrors.createError(SAP_ERROR_CODE, errorRecord);
+    public static void invokeOnBapiMessage(String output, Map<String, Resource> sapResource,
+                                           CallableUnitCallback callback, Context context) {
+        try {
+            Map<String, Object> properties = new HashMap<>();
```

Comment:
Can't we pass `null` for properties since we are not using it? Please check with standard library modules too.